### ISSUE TITLE
urql-svelte v2: Subscription is not updating changes from the db unless it is refreshed.

### ISSUE THREAD

**Issue by jarylozh** [FormidableLabs/urql#2473]
Hi,
I'm using urql v2 on svelte and I'm trying to get it to subscribe to my database via hasura. Currently, its pulling the data correctly BUT it is not updating whenever changes are made to the entries via mutations or direct modification.

**Source Code**
```
<script lang="ts">
	import type { GetAllDisasterSubSubscription } from '../generated/graphql';
	import { GetAllDisasterSubDocument } from '../generated/graphql';
	import { createClient as createWSClient } from 'graphql-ws';
	import {
		createClient,
		setContextClient,
		subscriptionStore,
		defaultExchanges,
		subscriptionExchange,
		debugExchange
	} from '@urql/svelte';

	import { WebSocket } from 'ws';

	const wsClient = createWSClient({
		webSocketImpl: WebSocket,
		url: 'ws://localhost:8080/v1/graphql'
	});

	const c = createClient({
		url: '/graphql',
		exchanges: [
			...defaultExchanges,
			debugExchange,
			subscriptionExchange({
				forwardSubscription: (operation) => ({
					subscribe: (sink) => ({
						unsubscribe: wsClient.subscribe(operation, sink)
					})
				})
			})
		]
	});

	setContextClient(c);

	$: subdisaster = subscriptionStore<GetAllDisasterSubSubscription>({
		client: c,
		variables: {},
		query: GetAllDisasterSubDocument
	});
</script>

<p class="text-white">Hello world</p>
<div class="bg-white-200 text-on-surface w-full h-full">
	{#if $subdisaster.data}
		{#if $subdisaster.error}
			<p>Oh no... {$subdisaster.error.message}</p>
		{/if}
		{#if $subdisaster.data}
			Fetching: {$subdisaster.fetching}
			<ul>
				{#each $subdisaster.data.disaster as message}
					<li>{message.summary}</li>
				{/each}
			</ul>
		{/if}
	{/if}
</div>
```

**Debug Exchanges**

![image](https://user-images.githubusercontent.com/49894234/171377611-1db3446d-c86b-4d1b-9052-8ad671b3f660.png)

Please advice. Thanks!

**Comment by frederikhors** [comment]
Maybe related to https://github.com/FormidableLabs/urql/discussions/2460?

**Comment by JoviDeCroock** [comment]
Should be fixed in #2514
